src/cli.c

```diff
@@ -0,0 +1,57 @@
+#include "cli.h"
+
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <stdio.h>
+
+#define MAX_HZ (1000000L)
+
+static const char* usage_str = "";
+
+__attribute__((__const__)) config_t* cli_config_default(void) {
+    config_t* conf = malloc(sizeof(config_t));
+
+    conf->debug = false;
+    conf->us_delay = 0;
+    conf->rom_path = "";
+
+    return conf;
+}
+
+void cli_config_destroy(config_t *conf) {
+    free(conf);
+}
+
+int cli_config_handle(config_t* const conf, int argc, char **argv) {
```

Comment:
I think that this function is missing some arguments sanity check and "correct arguments" check. I'm mentioning this since rom_path and delay are mandatory.